Sources/FirebaseFirestore/DocumentSnapshot+Swift.swift

```diff
@@ -8,7 +8,36 @@ import FirebaseCore
 import CxxShim
 import Foundation
 
+private typealias MapFieldValue = firebase.firestore.MapFieldValue
+
+// Expose these types as public to better match the types available from the Cocoa SDK.
 public typealias DocumentSnapshot = firebase.firestore.DocumentSnapshot
+public typealias ServerTimestampBehavior = firebase.firestore.DocumentSnapshot.ServerTimestampBehavior
+public typealias GeoPoint = firebase.firestore.GeoPoint
+public typealias FieldValue = firebase.firestore.FieldValue
+
+extension DocumentSnapshot {
+  public var reference: DocumentReference {
+    swift_firebase.swift_cxx_shims.firebase.firestore.snapshot_reference(self)
+  }
+
+  public func data(with behavior: ServerTimestampBehavior) -> [String: Any]? {
+    let data = swift_firebase.swift_cxx_shims.firebase.firestore.snapshot_get_data_workaround(self, behavior)
+
+    return FirestoreDataConverter.value(workaround: data)
+  }
+}
+
+extension DocumentSnapshot {
+  public func data<T: Decodable>(as type: T.Type,
+                                with serverTimestampBehavior: ServerTimestampBehavior = .none,
+                                decoder: Firestore.Decoder =  .init()) throws -> T {
```

Comment:
Indentation seems off?

```suggestion
  public func data<T: Decodable>(as type: T.Type,
                                 with serverTimestampBehavior: ServerTimestampBehavior = .none,
                                 decoder: Firestore.Decoder =  .init()) throws -> T {
```

Reply:
Isn't this going to screw up the indenting since it's not an even number of spaces?